tests/test_utilities/test_id_generator.py

```diff
@@ -0,0 +1,134 @@
+from moto.utilities.id_generator import (
+    TAG_KEY_CUSTOM_ID,
+    ExistingIds,
+    ResourceIdentifier,
+    Tags,
+    moto_id,
+    moto_id_manager,
+)
+
+ACCOUNT = "account"
+REGION = "us-east-1"
+RESOURCE_NAME = "my-resource"
+
+CUSTOM_ID = "custom"
+GENERATED_ID = "generated"
+TAG_ID = "fromTag"
+SERVICE = "test-service"
+RESOURCE = "test-resource"
+
+
+@moto_id
+def generate_test_id(
+    resource_identifier: ResourceIdentifier,
+    existing_ids: ExistingIds = None,
+    tags: Tags = None,
+):
+    return GENERATED_ID
```

Comment:
Bit of a nitpick, and not blocking a merge, but calling this `GENERIC_ID` may be more fitting, as that's a better opposite to `CUSTOM_ID`